analyses/infercnv-consensus-cell-type/exploratory-notebooks/SCPCP000004/01_nb-consensus-cell-types.Rmd

```diff
@@ -0,0 +1,238 @@
+  recursive = TRUE,
+  full.names = TRUE
+) |>
+  # set names as library id
+  purrr::set_names(
+    \(x) {
+      stringr::str_split_i(basename(x), "_", 1)
+    }
+  )
+
+# project sample metadata file
+metadata_file <- file.path(data_dir, "single_cell_metadata.tsv")
+
+# reference cell type groups
+reference_file <- file.path(reference_dir, "reference-cell-groups.tsv")
+```
+
+
+## Analysis
+
+```{r}
+# determine PDX libraries to exclude
+tumor_library_ids <- readr::read_tsv(metadata_file) |>
+  dplyr::filter(!is_xenograft) |>
+  dplyr::pull(scpca_library_id)
+
+# get cell groups
+cell_group_df <- readr::read_tsv(reference_file)
+
+# Prepare data frame of all cell types
+celltype_df <- celltype_files |>
+  # consider only non-PDX
+  purrr::keep_at(tumor_library_ids) |>
+  purrr::map(readr::read_tsv) |>
+  purrr::list_rbind() |>
+  dplyr::select(library_id, barcodes, consensus_annotation) |>
+  dplyr::left_join(cell_group_df, by = "consensus_annotation")
+```
+
+We'll first exclude any libraries that have fewer than 1000 cells from consideration in the reference.
+
+```{r}
+discard_libraries <- celltype_df |>
+  dplyr::count(library_id) |>
+  dplyr::filter(n < 1000) |>
+  dplyr::pull(library_id)
+
+discard_libraries
+```
+
+```{r}
+celltype_df <- celltype_df |>
+  dplyr::filter(!(library_id %in% discard_libraries))
+```
+
+These cell groupings are present in the neuroblastoma samples we're considering:
+
+```{r}
+unique(celltype_df$reference_cell_group)
+```
+
+We should remove any cell groups which might correspond to tumor cells, including `stromal`, `neuronal`, and `stem cell`.
+The `NA` values here correspond to `Unknown` cells, which we'll also remove.
+This means we'll keep groups `endothelial`, `epithelial`, and `immune`.
+
+```{r}
+exclude_groups <- c("stromal", "neuronal", "stem cell")
+celltype_df <- celltype_df |>
+  dplyr::filter(
+    !(reference_cell_group %in% exclude_groups),
+    consensus_annotation != "Unknown"
+  )
+```
+
+
+
+Let's begin by counting how many cells in each cell group we have per library
+
+```{r}
+celltype_count_df <- celltype_df |>
+  dplyr::count(library_id, reference_cell_group)
+
+# format for viewing
+celltype_count_wide_df <- celltype_count_df |>
+  tidyr::pivot_wider(
+    names_from = reference_cell_group,
+    values_from = n,
+    values_fill = 0
+  ) |>
+  # add column for total normal cells
+  dplyr::mutate(
+    all_normal = rowSums(dplyr::across(where(is.numeric)))
+  ) |>
+  # arrange by total normal cells
+  dplyr::arrange(desc(all_normal))
+
+# this way we can print the full table at once; RMarkdown will only show 10 rows at once
+knitr::kable(celltype_count_wide_df)
+```
+
+Virtually all of the normal cell types are `endothelial` and `immune`, with very few `adipocyte` and literally only one `epithelial` cell.
+
+We'll look at this data in barplot form to see, per-library, the relative proportions of each normal cell type group out of all normal cells in the library.
```

Comment:
I think we want to include a color for "other" to show us what percentage of each library is made up of the cells that we are interested in.